pilot/pkg/model/context.go

```diff
@@ -199,6 +202,17 @@ func (node *Proxy) SetServiceInstances(env *Environment) error {
 	return nil
 }
 
+func (node *Proxy) SetWorkloadLabels(env *Environment) error {
+	labels, err := env.GetProxyWorkloadLabels(node)
+	if err != nil {
+		log.Errorf("failed to get service proxy workload labels: %v", err)
+		return err
```

Comment:
I dont think you should bail here. We have had umpteen races where the workload connected to Pilot before Pilot could receive the pod info from kubernetes. You should default to using the workload labels supplied as part of the node metadata [I think its encoded as ISTIO_METAJSON_ or ISTIO_META_..].. cc @GregHanson who did some similar work for the Gateways. In fact the way this is written, it will undo the Gateway bug fix that @GregHanson did to solve some production issues.

Reply:
Sounds good. Will add that rescue.